File: angular2-client/src/client/app/model/objective-association.ts
import {CompanyObjective} from './company-objective';
import {DepartmentObjective} from './department-objective';

export interface ObjectiveAssociation {
  id: number;
  companyId: number;
  companyObjectiveId: number;
  departmentObjectiveId: number;
  companyObjective: CompanyObjective;
  departmentObjective: DepartmentObjective;
}

File: server/Service/Objectives/Dto/ObjectiveAssociation.cs
using Service.General.Dto;

namespace Service.Objectives.Dto
{
    public class ObjectiveAssociation : BaseDto, IDepartmentDto
    {
        public int Id { get; set; }

        public int CompanyId { get; set; }

        public int DepartmentId { get; set; }

        public int CompanyObjectiveId { get; set; }

        public int DepartmentObjectiveId { get; set; }

        public CompanyObjective CompanyObjective { get; set; }

        public DepartmentObjective DepartmentObjective { get; set; }
    }
}